lib/pharos/cluster_manager.rb

```diff
@@ -119,8 +121,8 @@ def apply_phases
 
       apply_phase(Phases::ConfigureBootstrap, [master_hosts.first]) # using `kubeadm token`, not the kube API
 
-      apply_phase(Phases::JoinNode, config.worker_hosts, parallel: true)
-      apply_phase(Phases::LabelNode, config.hosts, master: master_hosts.first, parallel: false) # NOTE: uses the @master kube API for each node, not threadsafe
+      apply_phase(Phases::JoinNode, config.worker_hosts)
+      apply_phase(Phases::LabelNode, [master_hosts.first], master: master_hosts.first)
```

Comment:
Labelnode now runs only on master instead of every host separately and goes through `@config.hosts` to get labels / taints